README.md

```diff
@@ -4,24 +4,39 @@ This is the repo for template deployment schemas hosted at https://schema.manage
-  1. [tools\validateJSON.js](tools/validateJSON.js) and [ResourceMetaSchema.json](tools/ResourceMetaSchema.json):  
-The script uses the ResourceMetaSchema.json to do some basic checks against the new/updated schema file.  
-**Usage:**   
-Node validateJSON.js \<schema file path\> ResourceMetaSchema.json \<schema folder path\>  
-**Sample:**  
-Node validateJSON.js ..\schemas\2015-08-01\Microsoft.Compute.json ResourceMetaSchema.json ..\schemas\
-  2. [tools\runSchemaTests.js](tools/runSchemaTests.js)  
-The script uses all test JSON files under [tests](tests/) folder to test against the schema files.  
-**Usage:**   
-Node runSchemaTests.js [--dir _folder_] [--AssertSubErrors]  
-&nbsp;&nbsp;&nbsp;&nbsp;***To run tests in single folder:***  
-&nbsp;&nbsp;&nbsp;&nbsp;Node runSchemaTests.js --dir ..\tests\2018-08-01  
-&nbsp;&nbsp;&nbsp;&nbsp;***To run tests in single folder and assert subErrors:***  
-&nbsp;&nbsp;&nbsp;&nbsp;Node runSchemaTests.js --dir ..\tests\2018-08-01 --AssertSubErrors  
+* Please add or update [tests](tests/) and run the full test suite before submitting your change (see [Running Tests](#running-tests))
+
+## Running Tests
+### Validating your schema adheres to the schema metaformat
+You can use [validateJSON.js](tools/validateJSON.js) against [ResourceMetaSchema.json](tools/ResourceMetaSchema.json) to do some basic checks against the new/updated schema file.
+#### Usage
+`node validateJSON.js <schema file path> ResourceMetaSchema.json <schema folder path>`
+#### Sample
+`node validateJSON.js ..\schemas\2015-08-01\Microsoft.Compute.json ResourceMetaSchema.json ..\schemas\`
+
+### Running the full set of schema tests
+You can use [runSchemaTests.js](tools/runSchemaTests.js) to test all the JSON files in [tests](tests/) against the schemas.
+#### Usage
+`node runSchemaTests.js [--dir _folder_] [--AssertSubErrors]`
+#### Samples
+##### Run tests in single folder
+`node runSchemaTests.js --dir ..\tests\2018-08-01`
+##### Run tests in single folder and assert subErrors
+`node runSchemaTests.js --dir ..\tests\2018-08-01 --AssertSubErrors`
+
+## Autogenerating from swagger
+You can use the generator in this repo to automatically generate a schema from a swagger spec checked into the [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) repo. We are working on fully automating this process, but please note that until then, it is your responsibility to ensure that the auto-generated schema has been correctly formatted before submitting a pull request.
+### Instructions
+1. Fork this repo, and clone it locally.
+2. Run the following commands (replace the provider namespace and api version accordingly):
+    * `cd generator`
+    * `npm install`
+    * `npm run generate-single -- -ProviderNamespace Microsoft.MyProvider -ApiVersion 2019-04-01`
+4. Review the generator logs to ensure no errors, and review the changes generated.
+5. Generate a commit and push it to your fork.
+6. Submit a pull request to this repo.
```

Comment:
this is cool. do we want partners to start using it? Are we going to stay in this stage for long time?

Reply:
I don't see why not. It may be useful for helping us identify bugs.